tests/Integration/Behaviour/Features/Context/Domain/AliasFeatureContext.php

```diff
@@ -216,67 +179,124 @@ public function assertSearchAliases(string $search, TableNode $tableNode): void
     }
 
     /**
-     * @BeforeFeature @restore-aliases-before-feature
+     * @Then I should get error that alias cannot be empty
+     * @Then I should get error that search term cannot be empty
+     *
+     * @return void
      */
-    public static function restoreAliasTables(): void
+    public function assertLastErrorIsInvalidAliasConstraint(): void
     {
-        DatabaseDump::restoreTables(['alias']);
+        $this->assertLastErrorIs(InvalidArgumentException::class);
```

Comment:
This exception is too generic, I guess it's triggered by the value objects? They should trigger a `AliasConstraintException` instead and use a specific code so we can differentiate which part is invalid the search term or the alias?

Reply:
It trigger a `PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException` in fact 🤔 
Maybe, we have an issue here. I will check that.

Reply:
![Capture d’écran 2024-11-21 à 11 42 54](https://github.com/user-attachments/assets/fb087bef-6694-4f73-b612-b0251a0f8449)

Reply:
Still not the right exception, did you find where it was thrown from?